### PR TITLE
atom-to-string scheme function

### PR DESCRIPTION
**Pull request by Dagiopia**
not sure if it's needed very much from your end or a similar functionality exists but I needed such a utility function and couldn't find one.

### CODE REVIEW

**Comment by ngeiswei** [comment]
I find strange we don't have this already, @linas?

Also, a better name would be
```
atom->string
```

**Comment by Dagiopia** [comment]
yes that's a better name.

File: opencog/guile/SchemeSmob.cc [hunk 1]
```diff
@@ -313,6 +313,7 @@ void SchemeSmob::register_procs()
 	register_proc("cog-number",            1, 0, 0, C(ss_number));
 	register_proc("cog-type",              1, 0, 0, C(ss_type));
 	register_proc("cog-arity",             1, 0, 0, C(ss_arity));
+	register_proc("atom->string",            1, 0, 0, C(ss_to_string));
```
Comment (linas) [inline opencog/guile/SchemeSmob.cc]: mis-matched whitespace indentation

**Comment by linas** [comment]
I don't think you need this function ... or rather, you can implement it in one line of scheme code as
```
(define (atom->string ATOM) (format #f "~A" ATOM))
```

Recall that `format` prints things (including atoms) and that `(format #t ...)` prints to stdout, and `(format #f ...)` prints to a string, and `(format DEV ...)` prints to an open file descriptor.

I'm closing this, because we don't need C++ code to do this; it already works

**Comment by linas** [comment]
FYI, the existing printing code is in `SchemeSmobGC.cc` from line 98 to the end-of-file.

**Comment by Dagiopia** [comment]
> I don't think you need this function ... or rather, you can implement it in one line of scheme code as
> 
> ```
> (define (atom->string ATOM) (format #f "~A" ATOM))
> ```
> 
> Recall that `format` prints things (including atoms) and that `(format #t ...)` prints to stdout, and `(format #f ...)` prints to a string, and `(format DEV ...)` prints to an open file descriptor.
> 
> I'm closing this, because we don't need C++ code to do this; it already works

`format` is just great! plus, unlike `oc_to_string` the id isn't printed wiith `format` 
sorry for the premature edit. will use this instead.

**Comment by linas** [comment]
no problem. Take a look at https://www.gnu.org/software/guile/manual/html_node/Formatted-Output.html for more.

**Comment by linas** [comment]
and in case you want to print to something other than a string, or to stdout, see https://www.gnu.org/software/guile/manual/html_node/Default-Ports.html

**Comment by ngeiswei** [comment]
BTW, the opencog logger supports format, so if you do
```scheme
(cog-logger-debug "my atom is ~a" atom)
```
it will log the content of `atom`.

**Comment by ngeiswei** [comment]
See https://wiki.opencog.org/wikihome/index.php/Scheme#Logger for more info.

**Comment by Dagiopia** [comment]
Thank you very much both! this is very helpful.

**Pull request closed without merging**